SCClassLibrary/Common/Control/Server.sc

```diff
@@ -511,6 +516,67 @@ Server {
 		^nodeAllocator.freePerm(id)
 	}
 
+	prHandleClientLoginInfoFromServer { |newClientID, newMaxLogins|
+
+		// turn notified off to allow setting clientID
+		statusWatcher.notified = false;
+
+		// only set maxLogins if not internal server
+		if (inProcess.not) {
+			if (newMaxLogins.notNil) {
+				if (newMaxLogins != options.maxLogins) {
+					"%: server process has maxLogins % - adjusting my options accordingly.\n"
+					.postf(this, newMaxLogins);
+				} {
+					"%: server process's maxLogins (%) matches with my options.\n"
+					.postf(this, newMaxLogins);
+				};
+				options.maxLogins = maxNumClients = newMaxLogins;
+			} {
+				"%: no maxLogins info from server process.\n"
+				.postf(this, newMaxLogins);
+			};
+		};
+
+		if (newClientID == clientID) {
+			"%: keeping clientID (%) as confirmed by server process.\n"
+			.postf(this, newClientID);
+		} {
+			"%: setting clientID to %, as obtained from server process.\n"
+			.postf(this, newClientID);
+		};
+		this.clientID = newClientID;
+		statusWatcher.notified = true; // and lock again
```

Comment:
@adcxyz - based on your description before:

> and is now precisely defined as "server is fully ready":
> booted, notified, final clientID, and server tree has run.

Isn't the correct place to set `notified = true` below in line 555 where it is set again?

Reply:
right, I did not remove it when I copied it to the right place! 
fixed, ran tests, and committed